faiss/impl/HNSW.cpp

```diff
@@ -641,6 +650,7 @@ int search_from_candidates(
                 if (dis < threshold) {
                     if (res.add_result(dis, idx)) {
                         threshold = res.threshold;
+                        nres += 1;
```

Comment:
This seems like a big change because it actually changes the return value, was it a bug? I don't see unit tests around the return value of this function, should we add them? Maybe it could also be a follow up.

Reply:
I think that this code fixes the bug. Please check the old baseline version of this function, https://github.com/facebookresearch/faiss/blob/691e01575e3a54c1c9094660f83415ab302d19d1/faiss/impl/HNSW.cpp#L610-L629 . It expects `nres` to grow